packages/react-components/react-nav-preview/src/components/NavItem/renderNavItem.tsx

```diff
@@ -13,6 +13,8 @@ export const renderNavItem_unstable = (state: NavItemState) => {
 
   return (
     <state.root>
+      {state.selected && state.selectedIcon && <state.selectedIcon />}
+      {!state.selected && state.unSelectedIcon && <state.unSelectedIcon />}
```

Comment:
I remember you mentioned if selected is undefined we don't render the unselected icon. If that's the case, this needs to be changed to:
```suggestion
      {state.selected !== undefined && state.icon && <state.icon />}
```

Reply:
I updated this now that we've only got one icon..... but we do want the icon rendering all the time if it's passed in.

Reply:
If there's no icon provided, probably not, or am I missing some context?

Reply:
er, I think we're okay here with the single icon implementation.

it's now just `{state.icon && <state.icon />}` which should be correct?

Reply:
should be okay, if there's no icon passed down, the slot won't render (taking care of my initial comment) and if passed we take care of the filled vs regular.